packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx

```diff
@@ -93,7 +93,9 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
                 <li key={index}>
                   {`${lookup[item.field!].headerName || item.field}
                   ${getOperatorLabel(item)}
-                  ${item.value ? getFilterItemValue(item) : ''}`}
+                  ${
+                    item.value !== undefined && item.value !== null ? getFilterItemValue(item) : ''
```

Comment:
Wouldn't this cover both `undefined` and `null` cases?
```suggestion
                    item.value != null ? getFilterItemValue(item) : ''
```

Reply:
I think implicit conversions are worse than explicit ones.

Reply:
You are right, in some cases like the comparison of numbers with strings, implicit comparisons might not work great (Example: 12 == "12" ✅) 

IMO though, for falsy values, using this shortcut could be a useful way to keep the code more readable especially if the intention is only to check for `undefined` or `null` (as no other falsy value like `0` or `false` would coerce to be `null`) 🤷‍♂️

Reply:
I'll use `item.value != null` with comment about it.